docs/architecture.md

```diff
@@ -0,0 +1,221 @@
+# Architecture
+
+## Introduction
+
+CRI Resource Manager (CRI-RM) controls how much and which resources are
+assigned to containers on a node in a Kubernetes cluster. In practice,
```

Comment:
In addition to Markus's point, I'd like to bring up quite early (maybe already here) also the fact that kubelet's CRI messaging is not the only event source. CRI-RM reacts to cluster configuration changes (cri-rm config, adjustments) and triggers actions completely independently (rebalance).

Reply:
I added a paragraph about configuration/configurability, but I think the part about how it is in practice an event source because it triggers a reconfiguration/rebalancing should only be part of the later sections. I think it'd be TMI without the necessary context to be properly understood at this point in the text.